test/unit/node/update.test.ts

```diff
@@ -23,6 +26,56 @@ describe("update", () => {
       return response.end(JSON.stringify(latest))
     }
 
+    if (request.url === "/reject-status-code") {
+      response.writeHead(500)
+      return response.end("rejected status code test")
+    }
+
+    if (request.url === "/no-location-header") {
+      response.writeHead(301)
+      response.writeHead(301, "testing", {
+        location: "",
+      })
+      return response.end("rejected status code test")
+    }
+
+    if (request.url === "/location-redirect") {
+      const latest: LatestResponse = {
+        name: "4.0.2",
+      }
+      response.writeHead(200)
+      return response.end(JSON.stringify(latest))
+    }
```

Comment:
Yeah, I think I did that originally but then something unexpected happened. My thinking was we shouldn't share `request.url` paths between tests because if you break one, it shouldn't break multiple tests, right?

Or do you think that's okay?

Reply:
Nevermind, I figured it out! Updating based on your suggestion